src/firebase-resource-manager.ts

```diff
@@ -0,0 +1,178 @@
+import * as clc from "cli-color";
+
+import * as api from "./api";
+import * as FirebaseError from "./error";
+import * as logger from "./logger";
+import { LroPoller, LroPollerOptions } from "./lro-poller";
+import { OraWrapper } from "./oraWrapper";
+
+const ONE_SECOND_MILLIS = 1000;
+
+export enum ParentResourceType {
+  ORGANIZATION = "organization",
+  FOLDER = "folder",
+}
+
+export interface ParentResource {
+  id: string;
+  type: ParentResourceType;
+}
+
+export class FirebaseResourceManager {
+  private poller: LroPoller = new LroPoller();
+
+  async createFirebaseProject(
+    projectId: string,
+    projectDisplayName: string,
+    parentResource?: ParentResource
+  ): Promise<{ projectNumber: string; projectId: string }> {
+    const cloudProjectCreationOperation = await this.createCloudProject(
+      projectId,
+      projectDisplayName,
+      parentResource
+    );
+    const { projectNumber } = await this.pollCloudProjectCreationOperation(
+      cloudProjectCreationOperation
+    );
+    const addFirebaseOperation = await this.addFirebaseToCloudProject(projectId);
+    const projectData = await this.pollAddFirebaseToCloudProjectOperation(addFirebaseOperation);
+
+    logger.info("");
+    logger.info("🎉🎉🎉 Your Firebase project is ready! 🎉🎉🎉");
+    logger.info("");
+    logger.info("Project information:");
+    logger.info(`   - Project ID: ${clc.bold(projectData.projectId)}`);
+    logger.info(`   - Project Name: ${clc.bold(projectData.displayName)}`);
+    logger.info("");
+    logger.info("Firebase console is available at");
+    logger.info(`https://console.firebase.google.com/project/${clc.bold(projectId)}/overview`);
+    return { projectNumber, projectId };
```

Comment:
Is the `projectNumber` needed for any downstream calls? The CLI typically uses `projectId` and not `projectNumber`, so I'm hoping the latter can be dropped.

Reply:
I agree that we can drop `projectNumber` now since we currently we are not using it. For `quickstart` commands it is required to enable Sign-in methods. We can add this back again in the future if we need to.

Reply:
Dropped `projectNumber` from the resolved object